code/modules/paperwork/paper.dm

```diff
@@ -299,9 +298,95 @@
 		img = P
 		user.drop_item(P, src)
 		user << "<span class='notice'>You attach the photo to the piece of paper.</span>"
+	else if(P.is_hot())
+		src.burn_paper(user)
+		return //no fingerprints, paper is gone
 	add_fingerprint(user)
+	return ..()
+
+/obj/item/proc/burn_paper(mob/user)
```

Comment:
I'd rather you renamed it so it could be used on anything.
Something like burn_item(). I've pointed out where you would need to change something for it.